modules/claims_api/app/controllers/claims_api/v1/application_controller.rb

```diff
@@ -91,6 +93,22 @@ def veteran_from_headers(with_gender: false)
         vet
       end
 
+      def validate_ccg_token!
+        client = TokenValidation::V2::Client.new(api_key: Settings.claims_api.token_validation.api_key)
+        root_url = request.base_url == 'http://localhost:3000' ? 'https://sandbox-api.va.gov' : request.base_url
+        claims_audience = "#{root_url}/services/claims"
+        request_method_to_scope = {
+          'GET' => 'claim.read',
+          'PUT' => 'claim.write',
+          'POST' => 'claim.write'
+        }
+
+        @is_token_valid ||= client.token_valid?(audience: claims_audience,
```

Comment:
I'm not good enough at ruby to know if this is a good case for an instance variable, but it seems like it?

my thought: since this gets called via `target_veteran` multiple times, there's no need to call the token service after the first time, so cache it in an instance variable

Reply:
I think this is fine.
If we didn't immediately raise for a `false` value, then there'd be some concern because on the 2nd/3rd time through, the `||=` would see a "falsey" value in the ivar and try to make the service call again.  But we are immediately raising in the case that the token is invalid, so this should be fine.